### PR TITLE
Fixing trivial ESLint checks

### PR DESCRIPTION
**Pull request by davidmsibley**
Fixing the grab bag of style misgivings, and pulling in the changes from master.

----

Contributor License Agreement adherence:

- [x] This Contribution is under the terms of Individual [Contributor License Agreements][] (and also Corporate Contributor License Agreements to the extent applicable) appearing in the [Apereo CLA roster][].

[Apereo CLA roster]: http://licensing.apereo.org/completed-clas
[Contributor License Agreements]: https://www.apereo.org/licensing/agreements

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -134,6 +134,17 @@
...
+      "angular/module-setter": "warn",
+      "angular/module-getter": "warn",
+      "angular/typecheck-object": "warn",
+      "angular/typecheck-array": "warn",
+      "angular/typecheck-string": "warn",
+      "angular/no-private-call": "warn",
+      "max-len": "warn",
+      "promise/catch-or-return": "warn",
+      "promise/always-return": "warn",
+
```
Comment (davidmsibley) [inline package.json]: These are the list of rules that have the most violations.  With the misc violations fixed, we can go through one by one to fix these.

File: package.json [hunk 2]
```diff
@@ -104,7 +104,7 @@
   "eslintConfig": {
     "root": true,
     "parserOptions": {
-      "ecmaVersion": 5
+      "ecmaVersion": 6
```
Comment (ChristianMurphy) [inline package.json]: What was causing errors parsing with the ES5 standard?

Comment (davidmsibley) [inline package.json]: almost everything is ES5 _except_ [tools/uw-frame-static/build.js](https://github.com/UW-Madison-DoIT/uw-frame/blob/master/tools/uw-frame-static/build.js)

Comment (davidmsibley) [inline package.json]: and unfortunately I can't change versions of the parser per file, it's a project level thing.

Comment (ChristianMurphy) [inline package.json]: Parser can be changed per folder.
an `.eslintrc.yml` or `eslintrc.json` file in a folder will override project level configuration.

Comment (ChristianMurphy) [inline package.json]: also configurations are merged, so only the `parserOptions` property would need to be set.

**Pull request merged**